src/Property.php

```diff
@@ -143,6 +143,41 @@ protected function cast($value)
         return new $castTo($value);
     }
 
+    protected function castArray(array $values)
+    {
+        foreach ($values as $value) {
+            if (! is_array($value)) {
+                return $this->cast($values);
+            }
+        }
+
+        $castTo = null;
+        $flatTypes = str_replace('[]', '', $this->types);
+
+        foreach ($flatTypes as $type) {
+            if (! is_subclass_of($type, DataTransferObject::class)) {
+                continue;
+            }
+
+            $castTo = $type;
+
+            break;
+        }
+
+        if (! $castTo) {
+            return $values;
+        }
+
+        $casts = array_map(
+            function ($value) use ($castTo) {
+                return new $castTo($value);
+            },
+            $values
+        );
+
+        return ! empty($casts) ? $casts : null;
```

Comment:
Empty casts means there were no items passed, right?

```php
$postData = new PostData([
    'tags' => []
]);
```

Or maybe even

```php
$postData = new PostData([]);
```

In the first case, I believe an empty array should be returned, not `null`. The second case should only be allowed if `tags` is nullable, but this might already work as is, you could test that.

Reply:
I believe the second case, as you said, does already work however I am unsure on the first. I should add additional test(s) and correct this if it's inconsistent with expected behavior.